rust/connlib/tunnel/src/client.rs

```diff
@@ -1571,13 +1571,6 @@ impl ClientState {
         // We remove all empty allowed ips entry since there's no resource that corresponds to it
         peer.allowed_ips.retain(|_, r| !r.is_empty());
 
-        // If there's no allowed ip left we remove the whole peer because there's no point on keeping it around
-        if peer.allowed_ips.is_empty() {
-            self.peers.remove(&gateway_id);
-            self.update_site_status_by_gateway(&gateway_id, ResourceStatus::Unknown);
-            // TODO: should we have a Node::remove_connection?
-        }
```

Comment:
@jamilbk I think this is what fixes #7761. Here, we update the site status of a gateway to "Unknown" if we disable a resource and this resource was last one that was "allowed" to send traffic. In other words, if you haven't used anything other than the Internet Resource and you toggle it, it will suddenly show "Unknown".

The tests detected that and it is fixed now. Please test on a real device.

Reply:
If this doesn't fix it then I am unsure what the bug is !